zenoh/tests/open_time.rs

```diff
@@ -0,0 +1,426 @@
+    time::{Duration, Instant},
+};
+
+use zenoh_config::Config;
+use zenoh_link::EndPoint;
+use zenoh_protocol::core::WhatAmI;
+
+const TIMEOUT_EXPECTED: Duration = Duration::from_secs(5);
+const SLEEP: Duration = Duration::from_millis(100);
+
+macro_rules! ztimeout_expected {
+    ($f:expr) => {
+        tokio::time::timeout(TIMEOUT_EXPECTED, $f).await.unwrap()
+    };
+}
+
+async fn time_open(
+    listen_endpoint: &EndPoint,
+    connect_endpoint: &EndPoint,
+    connect_mode: WhatAmI,
+    lowlatency: bool,
+) {
+    /* [ROUTER] */
+    let mut router_config = Config::default();
+    router_config.set_mode(Some(WhatAmI::Router)).unwrap();
+    router_config
+        .listen
+        .set_endpoints(vec![listen_endpoint.clone()])
+        .unwrap();
+    router_config
+        .transport
+        .unicast
+        .set_lowlatency(lowlatency)
+        .unwrap();
+    router_config
+        .transport
+        .unicast
+        .qos
+        .set_enabled(!lowlatency)
+        .unwrap();
+
+    let start = Instant::now();
+    let router = ztimeout_expected!(zenoh::open(router_config).into_future()).unwrap();
+    println!(
+        "open(mode:{}, listen_endpoint:{}, lowlatency:{}): {:#?}",
+        WhatAmI::Router,
+        listen_endpoint.as_str().split('#').next().unwrap(),
+        lowlatency,
+        start.elapsed()
+    );
+
+    /* [APP] */
+    let mut app_config = Config::default();
+    app_config.set_mode(Some(connect_mode)).unwrap();
+    app_config
+        .connect
+        .set_endpoints(vec![connect_endpoint.clone()])
+        .unwrap();
+    app_config
+        .transport
+        .unicast
+        .set_lowlatency(lowlatency)
+        .unwrap();
+    app_config
+        .transport
+        .unicast
+        .qos
+        .set_enabled(!lowlatency)
+        .unwrap();
+
+    /* [1] */
+    // Open a transport from the app to the router
+    let start = Instant::now();
+    let app = ztimeout_expected!(zenoh::open(app_config).into_future()).unwrap();
+    println!(
+        "open(mode:{}, connect_endpoint:{}, lowlatency:{}): {:#?}",
+        connect_mode,
+        connect_endpoint.as_str().split('#').next().unwrap(),
+        lowlatency,
+        start.elapsed()
+    );
+
+    /* [2] */
+    // Close the open transport on the app
+    let start = Instant::now();
+    ztimeout_expected!(app.close().into_future()).unwrap();
+    println!(
+        "close(mode:{}, connect_endpoint:{}, lowlatency:{}): {:#?}",
+        connect_mode,
+        connect_endpoint.as_str().split('#').next().unwrap(),
+        lowlatency,
+        start.elapsed()
+    );
+
+    /* [3] */
+    // Close the router
+    let start = Instant::now();
+    ztimeout_expected!(router.close().into_future()).unwrap();
+    println!(
+        "close(mode:{}, listen_endpoint:{}, lowlatency:{}): {:#?}",
+        WhatAmI::Router,
+        listen_endpoint.as_str().split('#').next().unwrap(),
+        lowlatency,
+        start.elapsed()
+    );
+
+    // Wait a little bit
```

Comment:
Why?

Reply:
@fuzzypixelz I believe it's considered common practice to have some delay after closing a session, we do this is all tests. This is to avoid issues like https://github.com/eclipse-zenoh/zenoh/issues/1019

Reply:
Yes. Btw it's copy pasted from there: https://github.com/eclipse-zenoh/zenoh/blob/912bd0c0fb728e2c66a40df0ec3b64d26239fc0b/io/zenoh-transport/tests/unicast_time.rs#L218